### ISSUE TITLE
Content changes for new subprojects Viz & datasets

### ISSUE THREAD

**Issue by stichbury** [kedro-org/kedro#2948]
## Description
When the subproject documentation is separated from the main framework project, there are some inevitable changes. Some will be straightforward link changes and others will be more significant since they impact the navigation between projects and ease of use of the entire set of docs.

I'm mindful that [#1474](https://github.com/kedro-org/kedro-viz/issues/1474) will have to take into account navigation but I don't want to couple the subproject work to a major overhaul of docs design and UX. For now, it's important to take the subproject work forward as this is a higher priority for us as a working team than the look and feel of the docs (from user research we know this isn't the highest priority).

So, for now, although things may not be pretty, we need to consider some options. I'll list out the various navigation issues when I've seen the subprojects setup and we can consider which are straightforward content changes and which require web dev experience to add elements on a (relatively temporary) basis until we redesign.

cc for info @astrojuanlu @tynandebold @stephkaiser

**Comment by stichbury** [comment]
> I'll list out the various navigation issues when I've seen the subprojects setup and we can consider which are straightforward content changes and which require web dev experience to add elements on a (relatively temporary) basis until we redesign.

So this is what I've considered as I've investigated the URL structure we'd get with the RTD subprojects, and as I understand it, we can use aliases to set up a set of docs with a shared prefix, so something like this (see https://docs.readthedocs.io/en/stable/subprojects.html#sharing-a-custom-domain for more).

```
* Kedro framework: https://docs.kedro.org/en/stable/. <--what we have now
* Kedro Viz: https://docs.kedro.org/projects/kedro-viz/en/stable/
* Datasets: https://docs.kedro.org/projects/kedro-datasets/en/stable/
```


But what does this look like in practice if a user navigates to the framework docs? 
* How do they find the Viz docs? 
* Does navigation to https://docs.kedro.org/projects/kedro-viz/en/stable/ mean that the Kedro framework docs ToC is unavailable?


What we ideally want the users to have access to is either: 

### 1. The Kedro docs embed a section of the ToC for Viz and Datasets

- Learn about Kedro
    - [Introduction to Kedro](https://docs.kedro.org/en/stable/introduction/index.html)
    - [First steps](https://docs.kedro.org/en/stable/get_started/index.html)

- Tutorial and basic Kedro usage
    - [Next steps: Tutorial](https://docs.kedro.org/en/stable/tutorial/spaceflights_tutorial.html)
    - [Visualisation with Kedro-Viz](https://docs.kedro.org/en/stable/visualisation/index.html)
    - [Experiment tracking in Kedro-Viz](https://docs.kedro.org/en/stable/experiment_tracking/index.html)
    - [Kedro for notebook users](https://docs.kedro.org/en/stable/notebooks_and_ipython/index.html)
    - etc

- Introduction to Kedro-Viz (this is stored at https://docs.kedro.org/projects/kedro-viz/en/stable/introduction.html)
    - Section of Viz docs WHEN YOU READ THIS THE TOC OPENS OUT BUT DOESN'T REPLACE THE TOC FOR FRAMEWORK
        - Page of Viz docs
        - Another page

- More Kedro docs in a section e.g. "Extend Kedro"
    -  Page (back to docs at https://docs.kedro.org/en/stable/extend_kedro/index.html)
    - Page

- Dataset docs (https://docs.kedro.org/projects/kedro-datasets/en/stable/introduction.html)
- Section of dataset docs WHEN YOU READ THIS SECTION THE TOC OPENS OUT BUT DOESN'T REPLACE THE TOC FOR FRAMEWORK
        - Page of dataset docs
        - Another page
        
- More Kedro docs in a section e.g. "Deploy Kedro"
    -  Page (back to docs at https://docs.kedro.org/en/stable/deployment/index.html)
    - Page

**Comment by stichbury** [comment]
@astrojuanlu @tynandebold I think this is the challenge, but let me know if there is a better way of looking at this (I can scribble some sketches) as I find it hard to visualise it.

**Comment by astrojuanlu** [comment]
This is what I had in mind (please excuse the hastily drawn diagrams):

![kedro-ng-docs](https://github.com/kedro-org/kedro/assets/316517/422f16fb-5c9f-4ca6-be5e-c214edf2a503)

So:

> But what does this look like in practice if a user navigates to the framework docs?
> - How do they find the Viz docs?

They're always visible on top

> - Does navigation to https://docs.kedro.org/projects/kedro-viz/en/stable/ mean that the Kedro framework docs ToC is unavailable?

Yes.

This is just one proposal, take it with a grain of salt.

**Comment by stichbury** [comment]
Ah, OK, nice. So addition of the new top nav is basically what this ticket needs to solve?

**Comment by astrojuanlu** [comment]
If we think this is a good navigation architecture, we can look into the technical feasibility, and I think Sphinx + RTD subprojects should work (I've seen other projects do similar things). When we're positive this is the direction we want to take, we probably need a hand from design to make this layout a reality, using any of the 3-column themes mentioned in https://github.com/kedro-org/kedro-viz/issues/1474.

**Comment by stichbury** [comment]
Exactly! I'm more than happy with moving the docs back into separate repos as I think that gives each project more autonomy in terms of release cadence. Also good because it makes the layout of the framework ToC shorter and more logical.

I am less enthusiastic about coupling in this work with https://github.com/kedro-org/kedro-viz/issues/1474 though because that's a big chunk of work/time and involves a lot more than a horizontal navigation bar. I'd like this to be quite separate otherwise it won't happen for ages because of the time required and research needed.

**Comment by astrojuanlu** [comment]
Good point. Let's give some thought at what the Minimum Viable Change looks like here.

**Comment by stichbury** [comment]
This will need to move into the next sprint as both @tynandebold and I are away from 28/08 - 04/09 but we should consider it then.
